src/lib.rs

```diff
@@ -14,147 +14,140 @@
-//! The use of `GeometryCollection` is discouraged, however.
-//!
-//! If your primary use case for this crate is ingesting `GeoJSON` strings in order to process geometries
-//! using the algorithms in the [`geo`](https://docs.rs/geo) crate, you can do so by enabling the `geo-types` feature in
-//! your `Cargo.toml`, and using the [`quick_collection`](fn.quick_collection.html) function to
-//! parse [`GeoJson`](enum.GeoJson.html) objects into
-//! a [`geo_types::GeometryCollection`](../geo_types/struct.GeometryCollection.html).
-//! See [here](#conversion-to-geo-objects) for details.
-//!
-//! Conversely, if you wish to produce a `FeatureCollection` from a homogenous collection of `geo` types, a `From` impl is
-//! provided for `geo_types::GeometryCollection`:
 //!
-//! ```rust
-//! # #[cfg(feature = "geo-types")]
-//! // The geo-types feature is required for this functionality
-//! # #[cfg(feature = "geo-types")]
-//! use geojson::FeatureCollection;
-//! # #[cfg(feature = "geo-types")]
-//! use geo_types::{GeometryCollection, LineString, Polygon};
-//! # #[cfg(feature = "geo-types")]
-//! use std::iter::FromIterator;
-//!
-//! # #[cfg(feature = "geo-types")]
-//! let poly = Polygon::new(
-//!     LineString::from(vec![(0., 0.), (1., 1.), (1., 0.), (0., 0.)]),
-//!     vec![],
-//! );
-//!
-//! # #[cfg(feature = "geo-types")]
-//! let polys = vec![poly];
-//! # #[cfg(feature = "geo-types")]
-//! let gc = GeometryCollection::from_iter(polys);
-//! # #[cfg(feature = "geo-types")]
-//! let fc = FeatureCollection::from(&gc);
-//! ```
+//! This crate helps you read and write [GeoJSON](https://geojson.org) — a format for encoding
+//! geographic data structures.
 //!
-//! This crate uses `serde` for serialization.
-//! To get started, add `geojson` to your `Cargo.toml`:
+//! To get started, add `geojson` to your `Cargo.toml`.
 //!
 //! ```text
 //! [dependencies]
-//! geojson= "*"
+//! geojson = "*"
 //! ```
+//!
+//! If you want to use GeoJSON as input to or output from a geometric processing crate like
+//! [`geo`](https://docs.rs/geo), see the section on [using geojson with
+//! geo-types](#use-geojson-with-other-crates-by-converting-to-geo-types).
+//!
+//! # Types and crate structure
+//!
+//! This crate is structured around the GeoJSON spec ([IETF RFC 7946](https://tools.ietf.org/html/rfc7946)),
+//! and users are encouraged to familiarise themselves with it.
+//!
+//! A GeoJSON object can be one of three top-level objects, reflected in this crate as the
+//! [`GeoJson`] enum members of the same name.
+//!
+//! 1. A [`Geometry`] represents points, curves, and surfaces in coordinate space.
+//! 2. A [`Feature`] usually contains a `Geometry` and some associated data, for example a "name"
+//!    field or any other properties you'd like associated with the `Geometry`.
+//! 3. A [`FeatureCollection`] is a list of one or more `Feature`s.
+//!
+//! Because [`Feature`] and [`FeatureCollection`] are more flexible, bare [`Geometry`] GeoJSONs are
```

Comment:
Very much down to individual style, but I usually see `GeoJSON is` rather than `GeoJSONs` are